packages/components/src/composite/stories/index.story.tsx

```diff
@@ -353,3 +354,50 @@ const Fill = ( { children } ) => {
 		},
 	},
 };
+
+/**
+ * Combining the `Tooltip` and `Composite` component has a few caveats.
+ * For example, the following code won't work as expected:
+ *
+ * ```jsx
+ * <Composite.Item
+ *   render={
+ *   <Tooltip text="Tooltip">
+ *     <button>Button</button>
+ *   </Tooltip>
+ * ```
+ *
+ * Take a look at this story's source code to see how to correctly compose
+ * these two components.
+ */
+export const WithTooltips: StoryObj< typeof Composite > = {
+	...Default,
+	args: {
+		...Default.args,
+		children: (
+			<>
+				<Tooltip text="Tooltip one">
+					<Composite.Item render={ <button /> }>
+						Item one
+					</Composite.Item>
+				</Tooltip>
+				<Composite.Item
+					render={ ( props ) => (
+						<Tooltip text="Tooltip two">
+							<button { ...props }>Item two</button>
+						</Tooltip>
+					) }
+				/>
+				<Composite.Item
+					render={ ( props ) => (
+						<Tooltip text="Tooltip three">
+							<button { ...props } />
+						</Tooltip>
+					) }
+				>
+					Item three
+				</Composite.Item>
```

Comment:
Is it better show all possible ways, or should we pick a recommended one and apply it to all composite items?

Reply:
Right now, it doesn't make it clear why we have three different ways to do it. I think it's helpful to have them all, but have some additional description / clarification - what to keep in mind and why.

Reply:
I'd keep only the first option. It's still unclear to me why the other two are necessary (why wouldn't you be able to wrap `<Composite.Item>`?). Unless there's a known use case for them, I'd exclude them for now.

Reply:
I feel the same as Diego here — I initially added all three versions, but my gut tells me that we should endorse the first option. It feels cleaner and more clear in terms of how the components compose with each other.

Reply:
Definitely no objections to keeping the first one 👍